resolver.go

```diff
@@ -31,16 +32,36 @@ type txtEntry struct {
 	expire time.Time
 }
 
-func NewResolver(url string) *Resolver {
+type Option func(*Resolver) error
+
+var NoMaxCacheTTL = -1
```

Comment:
This could be a `const`, and should not be exported. And with my comment below, we could get rid of it altogether.

Reply:
This value has now been removed.